my_sprites.py

```diff
@@ -207,11 +239,13 @@ def update(self):
 
         # searching state
         elif self.state == EnemyState.SEARCHING:
+            self.move_along_path()
             if not self.path:
                 self.state = EnemyState.ROAMING
 
         # roaming state
         elif self.state == EnemyState.ROAMING:
+            self.move_along_path()
             if not self.path:
```

Comment:
Let's add a comment here to explain what we do when there is no path, and have an _if..else_ so it clear that you get a new pos _or_ move along the path.